src/app/route/CheckoutPage/CheckoutPage.component.js

```diff
@@ -0,0 +1,625 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import ContentWrapper from 'Component/ContentWrapper';
+import Field from 'Component/Field';
+import Form from 'Component/Form';
+import ProductPrice from 'Component/ProductPrice';
+import Image from 'Component/Image';
+import CheckoutOrderSummary from 'Component/CheckoutOrderSummary';
+import './CheckoutPage.style';
+import TextPlaceholder from 'Component/TextPlaceholder';
+import { Link } from "react-router-dom";
+
+const CHECKOUT_STEP_SHIPPING = 'shipping';
+const CHECKOUT_STEP_REVIEW_AND_PAYMENTS = 'review-and-payments';
+
+
+class CheckoutPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            checkoutStep: CHECKOUT_STEP_SHIPPING, // shipping or review-and-payments
+
+            email: '',
+
+            shippingAddress: {
```

Comment:
My personal opinion: huge state declarations are confusing. Please try to get rid of this. Instead of this logic, we might keep it all Form component encapsulated.